cli/src/cmd/forge/script/broadcast.rs

```diff
@@ -237,9 +241,11 @@ impl ScriptArgs {
 
                 // Chains which use `eth_estimateGas` are being sent sequentially and require their
                 // gas to be re-estimated right before broadcasting.
-                if has_different_gas_calc(provider.get_chainid().await?.as_u64()) ||
-                    self.skip_simulation
+                if !is_fixed_gas_limit &&
+                    (has_different_gas_calc(provider.get_chainid().await?.as_u64()) ||
+                        self.skip_simulation)
                 {
+                    // TODO: Check that this case also works when setting gas limit for tx manually
```

Comment:
is this still a todo or covered via test?

Reply:
I added a test that should cover this and accordingly removed the TODO comment:
https://github.com/foundry-rs/foundry/pull/4219/commits/06b0bd04ad992ba62423b9c344854c2f31991b11